fpm/src/fpm_command_line.f90

```diff
@@ -249,6 +252,7 @@ subroutine get_command_line_settings(cmd_settings)
                  & with_executable=.true.,               &
                  & with_lib=.true.,                      &
                  & with_test=.true.,                     &
+                 & with_example=.true.,                  &
```

Comment:
I would actually prefer this to be false on plain `fpm new`.

Reply:
Per the discussions adding example/ to automatic discovery and build and adding a switch to run examples was to encourage their use.  Not creating them by default would be contrary to that. It is at least as easy to specify exactly what you do want on a `new` subcommand as it is is to add the --example command on `run`; and it is reasonable to surmise
the `new` command will be run infrequently  A new user should not get a surprise by three out of four being created. Since the auto-discovery has come a long way perhaps just removing the explicit entry for the demo program in the toml file so that removing the example/ directory does not require changing the manifest `fpm.toml` is a good idea; but I think a lot of packages I have seen over the years that were excellent utilities died away because of a lack of examples and documentation so I really `want` people to think about creating examples as a standard part of developing a package and putting a skeleton there by default is a good reminder. After deciding it should be automatically discovered and built it is very non-intuitive that it also not be created and populated by default.

Reply:
Changing the `fpm.toml` file an the names from main.f90 to something more descriptive would actually allow for any of the directories to be removed without repercussions. The auto-discovery takes care of that. Is it a good idea to have new users start with meta-data-only TOML file now that that is possible?  `fpm` can support a lot of simple projects now without anything else in there.  Most package managers have a `template` command or a number of skeleton repositories a user can pull for seeding more complex projects.  Perhaps nothing but the metadata and maybe a commented-out  remote and local dependency would get users on the right track. At some point the commented-out remote dependency might be for `stdlib`, or their could be a switch to add packages from a registered repository which seems inevitable at some point.